core/templates/pages/exploration-player-page/new-lesson-player/new-lesson-player-components/player-sidebar.component.ts

```diff
@@ -16,21 +16,86 @@
  * @fileoverview Component for the new lesson player sidebar
  */
 
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { downgradeComponent } from '@angular/upgrade/static';
+import { MobileMenuService } from '../new-lesson-player-services/mobile-menu.service';
 import './player-sidebar.component.css';
+import { ContextService } from 'services/context.service';
+import { I18nLanguageCodeService, TranslationKeyType } from
+  'services/i18n-language-code.service';
+import { ReadOnlyExplorationBackendApiService } from 'domain/exploration/read-only-exploration-backend-api.service';
+import { UrlService } from 'services/contextual/url.service';
 
 @Component({
   selector: 'oppia-player-sidebar',
   templateUrl: './player-sidebar.component.html',
   styleUrls: ['./player-sidebar.component.css'],
 })
-export class PlayerSidebarComponent {
-  isExpanded = false;
+export class PlayerSidebarComponent implements OnInit {
+  mobileMenuVisible: boolean = false;
+  isExpanded: boolean = false;
+  explorationId!: string;
+  expDesc!: string;
+  expDescTranslationKey!: string;
+  avgRating!: number | null;
+  fullStars: number = 0;
+  blankStars: number = 5;
+
+  constructor(
+    private mobileMenuService: MobileMenuService,
+    private contextService: ContextService,
+    private i18nLanguageCodeService: I18nLanguageCodeService,
+    private readOnlyExplorationBackendApiService:
+    ReadOnlyExplorationBackendApiService,
+    private urlService: UrlService,
+  ) {}
+
+  ngOnInit(): void {
+    this.mobileMenuService.getMenuVisibility().subscribe((visibility) => {
+      this.mobileMenuVisible = visibility;
+    });
+    let pathnameArray = this.urlService.getPathname().split('/');
+    let explorationContext = false;
+
+    for (let i = 0; i < pathnameArray.length; i++) {
+      if (pathnameArray[i] === 'explore' ||
+          pathnameArray[i] === 'create' ||
+          pathnameArray[i] === 'skill_editor' ||
+          pathnameArray[i] === 'embed' ||
+          pathnameArray[i] === 'lesson') {
+        explorationContext = true;
+        break;
+      }
+    }
+
+    this.explorationId = explorationContext ?
+      this.contextService.getExplorationId() : 'test_id';
+    this.expDesc = 'Loading...';
+    this.readOnlyExplorationBackendApiService.fetchExplorationAsync(
+      this.explorationId,
+      this.urlService.getExplorationVersionFromUrl(),
+      this.urlService.getPidFromUrl())
+      .then((response) => {
+        this.expDesc = response.exploration.objective;
+      });
+    this.expDescTranslationKey = (
+      this.i18nLanguageCodeService.
+        getExplorationTranslationKey(
+          this.explorationId, TranslationKeyType.DESCRIPTION)
+    );
+  }
 
   toggleSidebar(): void {
     this.isExpanded = !this.isExpanded;
```

Comment:
this.sidebarIsExpanded (some redundancy but don't start booleans with verbs)

Reply:
Done, sorry for the oversight